libparsec/crates/types/src/local_device.rs

```diff
@@ -50,6 +50,8 @@ impl LocalDevice {
         signing_key: Option<SigningKey>,
         private_key: Option<PrivateKey>,
         time_provider: Option<TimeProvider>,
+        user_realm_id: Option<VlobID>,
```

Comment:
I think this `generate_new_device` is too broad and we should instead replace it with multiple methods, each dedicated to a narrow usecase (especially since it naming is error prone, as we don't know if it means a new device for an existing or for a new user ^^).

`from_existing_device_for_user` is a good step in that direction (though I would rename this `new_device_for_user`)

So we can also add `new_user` that should be use when creating the first device for a user.

What do you think ?

Reply:
That makes sense.